src/electron/process_manager.ts

```diff
@@ -122,14 +133,20 @@ export class TunnelManager {
 
   constructor(private config: ShadowsocksConfig, private isAutoConnect: boolean) {
     this.routing = new RoutingDaemon(config.host || '', isAutoConnect);
+    this.shadowsocks = new SsLocal(config, SSLOCAL_PROXY_PORT);
+    this.tun2socks = new BadvpnTun2socks(SSLOCAL_PROXY_ADDRESS, SSLOCAL_PROXY_PORT);
```

Comment:
Pigybacking on @fortuna's comment, if instead TunnelManager had a `ipToShadowsocks` or `shadowsocksAdapter` member we inject at construction, it (1) makes the lifecycle management a little easier, and (2) pushes the implementation details of what processes exist to the different network stack code.  

It seems like the split processes are being taken advantage of in the connectivity test logic.  Maybe they can have a `performConnectivityTest()` function which adds the parameter to the start invocation Builder-style, or just `start` can take a `performConnectivityTest` argument?

Similarly, I guess the interface should have a `udpIsEnabled()` function